AK/Concepts.h

```diff
@@ -97,6 +107,14 @@ concept IteratorFunction = requires(Func func, Args... args)
     -> SameAs<IterationDecision>;
 };
 
+template<typename IteratorT, typename ContainedT>
+concept IteratorType = requires(IteratorT iterator, IteratorT end, ContainedT value)
```

Comment:
Why not use IteratorPairWih? All ranges should be usable with that, and there's no reason to check what type it returns for *it afaict

Reply:
IteratorPairWith only checks the iterator type, not the type that is returned by the iterator dereference

Reply:
why _should_ it care about the returned type? so long as it can be compared, it's all good right?

Reply:
Well because I'm not checking the comparator type anymore, I guess it's fine